### PR TITLE
debug: clean up opening of files in HyprCtl

### PR DESCRIPTION
**Pull request by WavyEbuilder**
`std::ifstream` is more suited than `execAndGet` here.

#### Describe your PR, what does it fix/add?

Works on part of https://github.com/hyprwm/Hyprland/issues/8400

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Just a quick note:

Regarding `result += "error";` for the case where `/proc/driver/nvidia/version` or `/etc/os-release` could not be opened, I'm attempting to match what `execAndGet` does:
```c++
// Execute a shell command and get the output
std::string execAndGet(const char* cmd) {
    CProcess proc("/bin/sh", {cmd});

    if (!proc.runSync())
        return "error";

    return proc.stdOut();
}
```
i.e. by appending `"error"` to `result`.

I don't mind adjusting this to be a bit more detailed or to print something else if that would be desired. 

#### Is it ready for merging, or does it need work?

Should be good other than the above.

### CODE REVIEW

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -949,11 +950,29 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
...
+                }
+            }
+            file.close();
+        } else
+            result += "error";
+    }
     result += "\n\n";
 
-    result += "os-release: " + execAndGet("cat /etc/os-release") + "\n\n";
+    std::ifstream file("/etc/os-release");
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:969]: this file is not closed properly (will linger until the end of the func), and it should be local probably.

```cpp
if (std::ifstream file("/etc/os-release"); file.is_open()) {
   // ...
}
```

File: src/debug/HyprCtl.cpp [hunk 2]
```diff
@@ -949,11 +950,29 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
...
 #endif
     result += "GPU information: \n" + GPUINFO;
-    if (GPUINFO.contains("NVIDIA") && std::filesystem::exists("/proc/driver/nvidia/version"))
-        result += execAndGet("cat /proc/driver/nvidia/version | grep NVRM");
+    if (GPUINFO.contains("NVIDIA") && std::filesystem::exists("/proc/driver/nvidia/version")) {
+        std::ifstream file("/proc/driver/nvidia/version");
+        std::string   line;
+        if (file.is_open()) {
+            while (std::getline(file, line)) {
+                if (line.find("NVRM") != std::string::npos) {
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:958]:
```cpp
if (!line.contains("NVRM"))
   continue;
```

File: src/debug/HyprCtl.cpp [hunk 3]
```diff
@@ -949,11 +950,29 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
...
+                    result += line;
+                    result += "\n";
+                }
+            }
+            file.close();
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:963]: redundant?

Comment (WavyEbuilder) [reply src/debug/HyprCtl.cpp]: yea, deconstructor calls it automatically. Sorted

File: src/debug/HyprCtl.cpp (hunk 2 above)
Comment (WavyEbuilder) [reply src/debug/HyprCtl.cpp]: Done

File: src/debug/HyprCtl.cpp (hunk 1 above)
Comment (WavyEbuilder) [reply src/debug/HyprCtl.cpp]: Done

**Review by vaxerski — APPROVED** [review]
thanks!

**Pull request merged**